tools/hls-verifier/lib/HlsVhdlTb.cpp

```diff
@@ -6,664 +6,574 @@
+    }
+
+    // Declare unused interfaces in the two port RAM
+    memInst.connect(D_OUT0_PORT, argName + "_" + D_OUT0_PORT);
+    memInst.connect(D_IN1_PORT, argName + "_" + D_IN1_PORT);
+    memInst.connect(WE1_PORT, argName + "_" + WE1_PORT);
+    memInst.connect(CE0_PORT, "\'1\'");
 
-  return "";
-}
+    memInst.emitVhdl(os);
+  }
 
-string HlsVhdlTb::getOutputFilepathForParam(const CFunctionParameter &param) {
-  if (param.isOutput)
-    return ctx.getVhdlOutPath(param);
+  void connectToDuv(Instance &duvInst) {
+    for (auto &[portSuffix, sigSuffix, bitwidth] : memrefToDPRAM)
+      duvInst.connect(argName + "_" + portSuffix, argName + "_" + sigSuffix);
+  }
+};
 
-  return "";
+// Centralizes the signal declaration for single argument models (both as inputs
+// and outputs)
+void declareSignalsSingleArgumentModel(mlir::raw_indented_ostream &os,
+                                       const std::string &argName,
+                                       unsigned dataWidth) {
+  // The single argument block needs to define all these signals, regardless
+  // of using as an input argument or an output argument
+  declareSTL(os, argName + "_" + CE0_PORT);
+  declareSTL(os, argName + "_" + WE0_PORT);
+  declareSTL(os, argName + "_din0", to_string(dataWidth));
+  declareSTL(os, argName + "_dout0", to_string(dataWidth));
+  declareSTL(os, argName + "_dout0_valid");
+  declareSTL(os, argName + "_dout0_ready");
 }
 
-// Declare a signal. Usage:
-// - declareSTL(os, "signal_name"); declares a std_logic signal
-// - declareSTL(os, "signal_name", "SIGNAL_WIDTH"); declares a std_logic_vector
-void declareSTL(mlir::raw_indented_ostream &os, const string &name,
-                std::optional<std::string> size = std::nullopt,
-                std::optional<std::string> initialValue = std::nullopt) {
-  os << "signal " << name << " : std_logic";
-  if (size)
-    os << "_vector(" << *size << " - 1 downto 0)";
-  if (initialValue)
-    os << " := " << *initialValue;
-  os << ";\n";
+void commonSingleArgumentDeclaration(Instance &inst,
+                                     const std::string &argName) {
+  inst.parameter(IN_FILE_PARAM, "INPUT_" + argName)
+      .parameter(OUT_FILE_PARAM, "OUTPUT_" + argName)
+      .parameter(DATA_WIDTH_PARAM, "DATA_WIDTH_" + argName)
+      .connect(CLK_PORT, "tb_" + CLK_PORT)
+      .connect(RST_PORT, "tb_" + RST_PORT)
+      .connect(DONE_PORT, "tb_temp_idle")
+      .connect(D_OUT0_PORT, argName + "_dout0")
+      .connect(D_OUT0_PORT + "_valid", argName + "_dout0_valid")
+      .connect(D_OUT0_PORT + "_ready", argName + "_dout0_ready");
 }
 
-// function to get the port name in the entitiy for each paramter
-void HlsVhdlTb::getConstantDeclaration(mlir::raw_indented_ostream &os) {
-  for (const auto &c : constants) {
-    os << "constant " << c.constName << " : " << c.constType
-       << " := " << c.constValue << ";\n";
+// A helper struct for connecting the in/output channels from/to the single
```

Comment:
I think this comment points to two structs. So I prefer "Helper structs for ..."